In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/wage2.csv')
# Описательные статистики по переменным модели
df[['IQ', 'south', 'urban', 'age', 'married', 'KWW']].describe()

IQ       south       urban         age     married         KWW
count  935.000000  935.000000  935.000000  935.000000  935.000000  935.000000
mean   101.282353    0.341176    0.717647   33.080214    0.893048   35.744385
std     15.052636    0.474358    0.450385    3.107803    0.309217    7.638788
min     50.000000    0.000000    0.000000   28.000000    0.000000   12.000000
25%     92.000000    0.000000    0.000000   30.000000    1.000000   31.000000
50%    102.000000    0.000000    1.000000   33.000000    1.000000   37.000000
75%    112.000000    1.000000    1.000000   36.000000    1.000000   41.000000
max    145.000000    1.000000    1.000000   38.000000    1.000000   56.000000

In [4]:
# специфицируем моделл через формулу
wage_eq1 = smf.ols(formula='np.log(wage)~IQ+south+urban+age+married+KWW', data=df).fit()
# Коэфициенты модели с округление
wage_eq1.params.round(3)

Intercept    5.126
IQ           0.007
south       -0.101
urban        0.165
age          0.014
married      0.191
KWW          0.007
dtype: float64

### Спецификация 
$$
log(wage)=5.126+0.007*IQ-0.101*south+ 0.165*urban+0.014*age+0.191*married+0.007*KWW
$$

### Интерпретация
1. При изменении результата теста IQ на 1, месячная зарплата увеличится на 7%, при прочих равных условиях.
2. При изменении восраста на год, месянчя зарплата увеличится на 14%, при прочих равных условиях.